elasticdl/python/common/tensor.py

```diff
@@ -47,12 +47,46 @@ def deserialize_tensor_pb(tensor_pb, tensor):
         )
     tensor.set(
         np.ndarray(shape=tensor_pb.dim, dtype=dtype, buffer=tensor_pb.content),
-        np.array(tensor_pb.indices),
+        np.array(tensor_pb.indices) if tensor_pb.indices else None,
         tensor_pb.name,
     )
     tensor_pb.Clear()
 
 
+def tensor_pb_to_ndarray(tensor_pb):
+    """Deserialize tensor protocol buffer and return a numpy ndarray."""
+    return Tensor.from_tensor_pb(tensor_pb).to_ndarray()
+
+
+def tensor_pb_to_tf_tensor(tensor_pb):
+    """Deserialize tensor protocol buffer and return a TensorFlow tensor."""
+    return Tensor.from_tensor_pb(tensor_pb).to_tf_tensor()
+
+
+def append_tensor_pb_from_ndarray(
```

Comment:
I think using emplace instead of append in this Api name will be more accurate.
emplace means construct and put the object at the container end without copy.

Reply:
Good point! Done.